src/data/list/basic.lean

```diff
@@ -552,8 +552,11 @@ by rw [concat_eq_append, reverse_append, reverse_singleton, singleton_append]
 @[simp] theorem reverse_reverse (l : list α) : reverse (reverse l) = l :=
 by induction l; [refl, simp only [*, reverse_cons, reverse_append]]; refl
 
+theorem reverse_involutive : involutive (@reverse α) :=
+λ l, reverse_reverse l
+
 theorem reverse_injective : injective (@reverse α) :=
```

Comment:
```suggestion
@[simp] theorem reverse_injective : injective (@reverse α) :=
```

Reply:
This fails CI with:
```
/- Checking 51416 declarations (plus 50261 automatically generated ones) in mathlib -/

/- The `simp_var_head` linter reports: -/
/- LEFT-HAND SIDE HAS VARIABLE AS HEAD SYMBOL.
Some simp lemmas have a variable as head symbol of the left-hand side: -/
-- data/list/basic.lean
#print list.reverse_injective /- Left-hand side has variable as head symbol: a₁ -/
```

Reply:
I believe this is a bug in the simp linter.  Just leave it out of the PR.  I'll try to add it in another PR with a fix to the linter.

Reply:
#4237